tests/lib/nested.sh

```diff
@@ -1230,6 +1225,60 @@ nested_exec_as() {
     sshpass -p "$PASSWD" ssh -p "$NESTED_SSH_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$USER"@localhost "$@"
 }
 
+nested_prepare_tools() {
+    if ! nested_exec "test -e retry" &>/dev/null; then
+        nested_copy "$TESTSTOOLS/retry"
+        nested_exec "sed 's/any-python/python3/g' -i ./retry"
+    fi
+
+    if ! nested_exec "test -e not" &>/dev/null; then
+        nested_copy "$TESTSTOOLS/not"
+    fi
+
+    if ! nested_exec "test -e MATCH" &>/dev/null; then
+        cat >> MATCH << 'EOF'
+#!/bin/bash
+set +xu
+stdin=$(cat)
+if ! echo "$stdin" | grep -q -E "$@"; then
+    res=$?
+    echo -e "grep error: pattern not found, got:\n$stdin" >&2
+    if [ "$res" != 1 ]; then
+        echo "unexpected grep exit status: $res"
+    fi
+    exit 1
+fi
```

Comment:
I'm concerned if this implementation of MATCH diverges from that of the rest of the codebase, can we more dynamically figure out the current state of the MATCH file implementation and copy that instead of writing this one in this helper sh file like this ?

Reply:
Thanks for reviewing this.
MATCH is defined in spread, we just use a fake function in the tools. Spread defines some functions like MATCH, NOMATCH REBOOT etc which are harcoded in any script executed in the remote system.
What I did is to copy the MATCH implementation from spread and deploy in the target system.

Reply:
Does it make sense for you? otherwise we can user 'grep -qE' instead

Reply:
I understand where MATCH comes from, I just would like to not introduce another implementation of MATCH here on the off chance that we find a bug with MATCH either in spread or in the version we use in snapd and then we forget about this implementation in nested.sh, and then months/years later we run into subtle bugs in nested VM's which rely on the fixed behavior of MATCH and we have to debug all over again why MATCH inside nested VM's is not operating the way we expect MATCH in normal spread runs to operate. 

Can you use `type MATCH` or something to do this? or just copy the version we use in snapd to the VM instead of introducing a copy here ?

Reply:
Nice, got it, thanks for the comment

Reply:
what you have now is good, thanks